### DISCUSSION TITLE
Better CDN support

### DISCUSSION THREAD

**Discussion by robsonsobral** [ExpressionEngine/ExpressionEngine#3267] [Feature Requests]
I had some discussion with a coworker about the throttling module. Today, it's something for an emergency only, because this kind of feature is something we manage at a different level of a site: in the front of the server. This also point to other anachronistic serving related things eecms has:

## Manipulated images always have the same name

The manipulated image files have the same name as the original filename, but on manipulation specific folders. Any modification can cause a cascade of stale content over the network. We could have partial MD5 hashes added to them to avoid that kind of issue.

## Assets management and cache busting

[Doug Black wrote an great article](https://u.expressionengine.com/article/compiling-web-assets-with-mix-and-webpack) about how to use [Laravel Mix](https://laravel-mix.com/) to compile assets, but, due to Mix own difficult to handle cache busting, the URLs are suffixed using a query string, which **isn't enough** for CDNs.

We are lucky! EECMS already has the solution for that: the [`{stylesheet="group/template}` path](https://docs.expressionengine.com/latest/templates/globals/stylesheet.html) tag! I'm going back to eecms after years of JS projects and I never imagined I would use `{stylesheet}` again. However, the build process become so complex that any help are welcomed. Over the time, I used Gulp, Parcel, webpack, pure NPM scripts and more tools than I can remember now. We can use Laravel Mix or whatever to compile/build the assets and just dump the final files at a "template.group" folder.

Could we have a `{script="group/template"}` path tag? Then, we could add the hash suffix to that!

## Search by POST and non cacheable URLs

Now, the part that I think it is the harder to improve: the search results templates.

- they aren't sharable;
- they aren't cacheable;

To fix that, the `{exp:search:search_results}` tag needs to be able to use parameters from URL. Using GET, the query string parameters also need to be alphabetically sorted, to make sure the full URL is always the same.

There's a legend that changes are being made to the search form. I'm curious to see them!

**Comment by jcogs-design** [comment]
Re **Assets management and cache busting**

Adding a `{script}` tag to mirror the `{stylesheet}` tag is understandable: the `{stylesheet}` tag primarily offers a route to **avoid** template parsing of the content of the stylesheet and so some performance gain when serving the file from within EE, and I guess a similar benefit could be obtained for javascript files.

However in your block quote comment you focus on the description in the documentation of an option to explicitly run the stylesheet through the parsing process - which is fine if this is helpful - but the solution offered (to link to the file concerned using the `{path=}` global template variable) is one that could equally well be applied to javascript files too.  Which suggests the benefit you are seeking is already provided...?

I am guessing I'm missing something in this... hence request for clarification.

**Comment by robsonsobral** [comment]
@jcogs-design , I'm sorry! I wrote on my code editor, pasted here and edited again. I'm under the stress of a huge project right now.

I removed the nonsense part. I'm sorry. And thank you for pointing me to the confusing part.

**Comment by robsonsobral** [comment]
(See #1230 for how to make CDN content trustful.)

**Comment by intoeetive** [comment]
@robsonsobral now that we have merged #1259, should we consider this as done and close the issue?

**Comment by robsonsobral** [comment]
Per topic:

**Manipulated images always have the same name**

I've no idea. I haven't tested eecms 7 yet.

**Assets management and cache busting**

Yes! It's done!

**Search by POST and non cacheable URLs**

I don't know. I haven't tested the Search Pro module.

**Comment by intoeetive** [comment]
The manipulations are based on the filename indeed. The reason for this is that those are not being recorded in the database, so the original file name plus manipulation name is really the only way to find the manipulated file currently.

Keeping this topic open is anyone wants to share more ideas on how we can do this better!
